flow/layers/physical_shape_layer.cc

```diff
@@ -51,11 +51,11 @@ void PhysicalShapeLayer::Preroll(PrerollContext* context,
     set_needs_system_composite(true);
 #else
     // Add some margin to the paint bounds to leave space for the shadow.
-    // The margin is hardcoded to an arbitrary maximum for now because Skia
+    // The margin is set to a multiple of elevation_ for now because Skia
     // doesn't provide a way to calculate it.  We fill this whole region
     // and clip children to it so we don't need to join the child paint bounds.
     SkRect bounds(path_.getBounds());
-    bounds.outset(20.0, 20.0);
+    bounds.outset(elevation_ * 1.5, elevation_ * 1.5);
```

Comment:
The following outset might be more accurate according to the physical model used by Flutter and Skia:
```
ex = (kLightRadius + bounds.width() * 0.5) / kLightHeight;
ey = (kLightRadius + bounds.height() * 0.5) / kLightHeight;
bounds.outset(elevation_ * ex, elevation_ * ey);
```
See 
![img_20190305_143908](https://user-images.githubusercontent.com/22987568/53842666-0169ed00-3f55-11e9-9d47-e5179188d506.jpg)
for the illustration.

Reply:
BTW, be careful about `device_pixel_ratio_`. It seems that all `elevation_, kLightRadius, kLightHeight` should be multiplied with `device_pixel_ratio_`. The ratio of `elevation_` and `kLightHeight` cancels out, but you'll still need `kLightRadius * device_pixel_ratio_`.

Reply:
I've updated it and tried to incorporate your diagram as a comment.  Does that look right?